src/main/java/com/example/onlinebookstore/repository/impl/BookRepositoryImpl.java

```diff
@@ -0,0 +1,64 @@
+package com.example.onlinebookstore.repository.impl;
+
+import com.example.onlinebookstore.exception.EntityNotFoundException;
+import com.example.onlinebookstore.model.Book;
+import com.example.onlinebookstore.repository.BookRepository;
+import java.util.List;
+import java.util.Optional;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
```

Comment:
You don't need Autowired here as you have one constructor
Also, be consistent and you can use RequiredArgsConstructor annotation like in BookServiceImpl